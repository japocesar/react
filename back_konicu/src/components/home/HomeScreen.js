
import React, { useEffect } from 'react';

import { Circle } from '../ui/circle/Circle';
import { bannerCircles } from '../../data/bannerCircles';
import { cards } from "../../data/cards";
import { home } from "../../helpers/parallax";


import "animate.css";
import './home.scss';
import { Card } from '../ui/card/Card';



export const HomeScreen = () => {

    useEffect(() => {
        home();
    }, [])
    
    return (
    <>
       <section>
        <div className="row justify-content-md-center m-0">
            <div className="col-12 col-offset-md-2 col-md-8 p-0 img-container"></div>
        </div>

        <div className="row m-0 justify-content-around circles-container">
            { bannerCircles.map( act => 
                (
                    <Circle key={ act.texto } { ...act } />
                )
            )}
        </div> 
       </section>

        <section  id="cards" className="panel orange align-top my-3">
            <div className="container cards-container">
                <div className="card-deck container">
                    { cards.map( card => (
                        <Card key={ card.titulo } { ...card } />
                    ) )}
                </div>
            </div>
        </section>

        {/* <section style={{ height: 300 }} id="sponsors" className="panel red align-top">
            <p id="sponsors-content"><code>start: "top center"</code> pins this element when the <strong>top</strong> of the red element hits the <strong>center</strong> of the viewport, and remains pinned for 200px because its <code>end</code> is defined as <code>"+=200"</code></p>
        </section>

        <section style={{ height: 300 }} className="panel purple">
            Easy Peasy!
        </section> */}

    </>

    )
}
