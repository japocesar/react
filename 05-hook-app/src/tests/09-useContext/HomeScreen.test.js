import React from 'react'
import { mount } from 'enzyme';
import { HomeScreen } from '../../components/09-useContext/HomeScreen';
import { UserContext } from '../../components/09-useContext/UserContext';

describe('Tests in <HomeScreen />', () => {
    const user = {
        name: 'Cesar',
        email: 'test@test.com'
    }
    const wrapper = mount(
        <UserContext.Provider value={{
            user,
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('should display the component', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    
})