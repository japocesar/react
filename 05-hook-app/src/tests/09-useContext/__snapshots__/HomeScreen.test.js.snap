// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <HomeScreen /> should display the component 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Cesar",
   "email": "test@test.com"
}
  </pre>
</div>
`;
